core/xpdo/compression/xpdozip.class.php

```diff
@@ -209,4 +220,32 @@ public function hasError() {
     public function getErrors() {
         return $this->_errors;
     }
+
+    /**
+     * Check that the filename has a file type extension that is allowed
+     *
+     * @param $filename
+     * @return bool
+     */
+    private function checkFiletype($filename)
+    {
+        if ($this->xpdo->getOption('allowedFileTypes')) {
+            $allowedFileTypes = $this->getOption('allowedFileTypes');
+            $allowedFileTypes = (!is_array($allowedFileTypes)) ? explode(',', $allowedFileTypes) : $allowedFileTypes;
+        } else {
+            $allowedFiles = $this->xpdo->getOption('upload_files') ? explode(',', $this->xpdo->getOption('upload_files')) : array();
+            $allowedImages = $this->xpdo->getOption('upload_images') ? explode(',', $this->xpdo->getOption('upload_files')) : array();
```

Comment:
the second getOption here as `upload_files` instead of `upload_images`

Reply:
Oh, that must be wrong on the original code too.

Reply:
Gotcha: https://github.com/modxcms/revolution/blob/bcb5ab3eac04dcdd834db5af5b775c011cd82385/core/model/modx/sources/modfilemediasource.class.php#L576